### ISSUE TITLE
CancellationTokenSource in `CreateContext` doesn't do anything.

### ISSUE THREAD

**Issue by veleek** [Z00L/ReswFileCodeGenerator#2]
In the auto generated `CreateContext` method, a `CancellationToken` is passed to the `AsTask` extension.  Cancelling the token source doesn't do anything unless you're `Wait()`ing on the task, and since you only cancel after you've waited it will have no effect.

Yes, the seeming result of that call is that the runtime will actually cancel the thread, but it doesn't actually do so.  The actual `dispatcher.RunAsync` call will continue to completion regardless of what you do.

**Comment by veleek** [comment]
I would be willing to fix this issue, but just wanted to throw it up to track in case I don't get back to it.

**Comment by Z00L** [comment]
I think you are saying line `cancellationTokenSource.Cancel();` is useless as the queued task (`() => context = ResourceContext.GetForCurrentView()`) on target dispatcher is executed nevertheless. I have not tested this, so you could be right. However this line does not cause any crash, does it? Do you recommend to just strip this line? But I'm not really sure if I got you.
```C#
var context = default(ResourceContext);
var action = dispatcher.RunAsync(CoreDispatcherPriority.High, () => context = ResourceContext.GetForCurrentView());
var task = WindowsRuntimeSystemExtensions.AsTask(action, cancellationTokenSource.Token);
// Runs into a 'timeout', when the 'main-thread' is blocked recently.
// Don't wait too long as there is an alternative way to retrieve an context (for independent use).
task.Wait(TimeSpan.FromSeconds(1));
cancellationTokenSource.Cancel();
if ((context != null))
{
    return context;
}
```

**Comment by veleek** [comment]
Correct, no crash, it just doesn't do anything.  I would just remove the CancellationTokenSource completely.

Another option (that would require you to have a try/catch) would be:

```csharp
using(CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(1)))
{
  try
  {
    RunAsync().AsTask(cts.Token)
  }
  catch(TaskCancelledException)
  {
    // Ignore the cancelled task.  
  }

  return if not null;
}
```

That way you still have the CancellationTokenSource if you need it later.  However I don't think that's better than just removing it and using the `.Wait()` with a timeout like you already have.
